### ISSUE TITLE
`prefer-logical-operator-over-ternary`: Handle boolean literal ternaries

### ISSUE THREAD

**Issue by Sli4o** [sindresorhus/eslint-plugin-unicorn#3704]
### Description

`prefer-logical-operator-over-ternary` does not currently report ternaries whose branches are boolean literals, even when they can be expressed more directly as boolean/logical expressions.

This becomes especially noticeable together with `prefer-boolean-return`, which can turn multiple boolean guard returns into a ternary that is still unnecessarily verbose.

Tested with eslint-plugin-unicorn v75.

### Example

```ts
class X {
	private isAbc = true;
	private value = '123';

	private abc() {
		return this.isAbc ? true : false;
	}

	private xyz() {
		return this.value ? true : false;
	}

	private isSubPath(pathA: string, pathB: string) {
		if (pathA === '/') {
			return true;
		}

		if (pathA === pathB) {
			return true;
		}

		return pathA.startsWith(pathB) && pathA[pathB.length] === '/';
	}
}
```

After applying the existing `prefer-boolean-return` fix, `isSubPath` becomes equivalent to:

```ts
private isSubPath(pathA: string, pathB: string) {
	return pathA === '/' || pathA === pathB
		? true
		: pathA.startsWith(pathB) && pathA[pathB.length] === '/';
}
```

The resulting ternary is not reported by `prefer-logical-operator-over-ternary`.

### Expected behavior

The boolean literal ternaries could be simplified to direct boolean/logical expressions:

```ts
class X {
	private isAbc = true;
	private value = '123';

	private abc() {
		return this.isAbc;
	}

	private xyz() {
		return !!this.value;
	}

	private isSubPath(pathA: string, pathB: string) {
		return (
			pathA === '/'
			|| pathA === pathB
			|| (pathA.startsWith(pathB) && pathA[pathB.length] === '/')
		);
	}
}
```

In general:

```js
condition ? true : false
```

can become a boolean coercion of `condition`, while:

```js
condition ? true : expression
```

when `expression` is boolean, can become:

```js
condition || expression
```

This would also make the output of `prefer-boolean-return` compose more cleanly with `prefer-logical-operator-over-ternary`.
